pkg/internalapis/featurestates/featurestates.go

```diff
@@ -0,0 +1,368 @@
+		newfssConfigMap.Namespace == supervisorFeatureStateConfigMapNamespace {
+		if reflect.DeepEqual(newfssConfigMap.Data, oldfssConfigMap.Data) {
+			return
+		}
+		var err error
+		var featureStates []featurestatesv1alpha1.FeatureState
+		for feature, state := range newfssConfigMap.Data {
+			var featureState featurestatesv1alpha1.FeatureState
+			featureState.Name = feature
+			featureState.Enabled, err = strconv.ParseBool(state)
+			if err != nil {
+				log.Errorf("failed to parse feature state value: %v for feature: %q", state, feature)
+				return
+			}
+			featureStates = append(featureStates, featureState)
+		}
+		for {
+			// if failed to create or update CR in the namespace, then error is not returned
+			// for such cases, CR update will be enqueued to pendingCRUpdatesObj
+			err := updateFeatureStatesCRInAllWorkLoadNamespaces(ctx, featureStates)
+			if err != nil {
+				log.Errorf("failed to update feature states CR in the all workload namespaces. err: %v", err)
+			}
+		}
+	}
+}
+
+// configMapAdded adds feature state switch values from configmap that has been created on K8s cluster
+func namespaceAdded(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	namespace, ok := obj.(*v1.Namespace)
+	if namespace == nil || !ok {
+		log.Warnf("namespaceAdded: unrecognized object %+v", obj)
+		return
+	}
+	if _, ok = namespace.Labels[WorkLoadNamespaceLabelKey]; ok {
+		log.Infof("observed new workload namespace: %v", namespace.Name)
+		for {
+			// if failed to create CR in the namespace, then error is not returned
+			// for such case, CR update will be enqueued to pendingCRUpdatesObj
+			err := createFeatureStatesCRInWorkLoadNamespace(ctx, namespace.Name)
+			if err == nil {
+				break
+			}
+			log.Errorf("failed to create feature states CR in the workload namespace: %q, err: %v", namespace, err)
+		}
+	}
+}
+
+// getFeatureStates returns latest feature states from supervisor
+func getFeatureStates(ctx context.Context, k8sClient clientset.Interface) ([]featurestatesv1alpha1.FeatureState, error) {
+	log := logger.GetLogger(ctx)
+	//  Retrieve SV FeatureStates configmap
+	fssConfigMap, err := k8sClient.CoreV1().ConfigMaps(supervisorFeatureStateConfigMapNamespace).Get(ctx, supervisorFeatureStatConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("failed to retrive SV featureswitch state from namespace:%q with name: %q", supervisorFeatureStateConfigMapNamespace, supervisorFeatureStatConfigMapName)
+		return nil, err
+	}
+
+	var featureStates []featurestatesv1alpha1.FeatureState
+	for feature, state := range fssConfigMap.Data {
+		var featureState featurestatesv1alpha1.FeatureState
+		featureState.Name = feature
+		featureState.Enabled, err = strconv.ParseBool(state)
+		if err != nil {
+			log.Errorf("failed to parse feature state value: %v for feature: %q", state, feature)
+			return nil, err
```

Comment:
As this function indirectly gets called in `StartSvFSSReplicationService`, should we error out if the value is not correct or just mark the feature as false? I guess our current code marks a feature state as false if the value is not bool

Reply:
done. instead of erroring out, when parsing fails, marking the feature as false.